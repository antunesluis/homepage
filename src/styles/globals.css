@tailwind base;
@tailwind components;
@tailwind utilities;

:root {
  /* Light theme */
  --color-background: #fafafa;
  --color-foreground: #18181b;

  /* Primary colors - Blue/Indigo palette */
  --color-primary: #4f46e5;
  --color-primary-hover: #4338ca;
  --color-primary-muted: #e0e7ff;

  /* Secondary colors - Teal palette */
  --color-secondary: #0d9488;
  --color-secondary-hover: #0f766e;
  --color-secondary-muted: #ccfbf1;

  /* Accent colors */
  --color-accent: #f43f5e;
  --color-accent-hover: #e11d48;
  --color-accent-muted: #ffe4e6;

  /* Gray scale */
  --color-muted: #f4f4f5;
  --color-border: #e4e4e7;
  --color-border-hover: #d4d4d8;

  /* Surface colors for cards and elevated elements */
  --color-surface: #ffffff;
  --color-surface-hover: #f8fafc;
}

.dark {
  /* Dark theme */
  --color-background: #09090b;
  --color-foreground: #fafafa;

  /* Primary colors - adjusted for dark mode */
  --color-primary: #6366f1;
  --color-primary-hover: #818cf8;
  --color-primary-muted: #1e1b4b;

  /* Secondary colors - adjusted for dark mode */
  --color-secondary: #14b8a6;
  --color-secondary-hover: #2dd4bf;
  --color-secondary-muted: #134e4a;

  /* Accent colors - adjusted for dark mode */
  --color-accent: #fb7185;
  --color-accent-hover: #fda4af;
  --color-accent-muted: #4c1d24;

  /* Gray scale - darker variants */
  --color-muted: #27272a;
  --color-border: #3f3f46;
  --color-border-hover: #52525b;

  /* Surface colors for dark mode */
  --color-surface: #18181b;
  --color-surface-hover: #27272a;
}

body {
  @apply bg-background text-foreground;
  transition:
    background-color 0.3s ease-in-out,
    color 0.3s ease-in-out;
}
