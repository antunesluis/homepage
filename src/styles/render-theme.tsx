import { render, RenderResult } from '@testing-library/react';
import { ThemeProvider } from 'next-themes';

export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(
    <ThemeProvider attribute="class" defaultTheme="light" enableSystem={false}>
      {children}
    </ThemeProvider>,
  );
};
