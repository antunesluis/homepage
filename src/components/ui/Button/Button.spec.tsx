import { screen } from '@testing-library/react';
import { renderTheme } from '@/styles/render-theme';
import { Button } from '.';

describe('<Button />', () => {
  it('renders correctly with default props', () => {
    renderTheme(<Button>Test</Button>);

    const button = screen.getByRole('button', { name: /test/i });
    expect(button).toBeInTheDocument();
  });

  it('renders with different variants', () => {
    const { rerender } = renderTheme(<Button variant="primary">Test</Button>);
    expect(screen.getByRole('button')).toHaveClass('bg-primary');

    rerender(<Button variant="secondary">Test</Button>);
    expect(screen.getByRole('button')).toHaveClass('bg-secondary');

    rerender(<Button variant="outline">Test</Button>);
    expect(screen.getByRole('button')).toHaveClass('border-border');
  });

  it('applies size styles correctly', () => {
    const { rerender } = renderTheme(<Button size="sm">Test</Button>);
    expect(screen.getByRole('button')).toHaveClass('h-8 px-3 text-sm');

    rerender(<Button size="md">Test</Button>);
    expect(screen.getByRole('button')).toHaveClass('h-10 px-4 text-sm');

    rerender(<Button size="lg">Test</Button>);
    expect(screen.getByRole('button')).toHaveClass('h-12 px-6 text-base');
  });

  it('renders with icons', () => {
    const leftIcon = <span data-testid="left-icon" />;
    const rightIcon = <span data-testid="right-icon" />;
    renderTheme(
      <Button leftIcon={leftIcon} rightIcon={rightIcon}>
        With Icons
      </Button>,
    );

    expect(screen.getByTestId('left-icon')).toBeInTheDocument();
    expect(screen.getByTestId('right-icon')).toBeInTheDocument();
  });

  it('should match the snapshot', () => {
    const { container } = renderTheme(<Button />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
