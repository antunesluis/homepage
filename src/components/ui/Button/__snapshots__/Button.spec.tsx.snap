// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Button /> > should match the snapshot 1`] = `
<button
  class="inline-flex items-center justify-center font-medium transition-colors duration-200 rounded-lg select-none focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 active:translate-y-[1px] bg-primary hover:bg-primary-hover text-white focus-visible:ring-primary/50 h-10 px-4 text-sm gap-2"
  type="button"
/>
`;
